### PR TITLE
Rename --checked to --check-asserts

### PR DESCRIPTION
**Pull request by natebosch**
Keep the old flag as an alias, run with checked mode if either is set.
Behavior will be weird for cases like `--checked --no-check-asserts` but
it's not worth the extra complexity to track that down.

Update tests using the flag.

Use consistent single quotes in the files that were being edited anyway.

### CODE REVIEW

**Comment by kevmoo** [comment]
Shouldn't we be going with `--enable-asserts` ? To be consistent with...everything else?

**Comment by natebosch** [comment]
See https://github.com/dart-lang/pub/issues/1922#issuecomment-398352483

**Comment by natebosch** [comment]
discussed offline. Went with `--enable-asserts` and update the help string to match `dart --help`

File: lib/src/command/global_run.dart [hunk 1]
```diff
@@ -13,28 +13,28 @@ import '../utils.dart';
...
       "NOTE: We are currently optimizing this command's startup time.";
-  String get invocation => "pub global run <package>:<executable> [args...]";
+  String get invocation => 'pub global run <package>:<executable> [args...]';
   bool get allowTrailingOptions => false;
 
   GlobalRunCommand() {
-    argParser.addFlag("checked",
-        abbr: "c", help: "Enable runtime type checks and assertions.");
-    argParser.addOption("mode", help: "Deprecated option", hide: true);
+    argParser.addFlag('enable-asserts', help: 'Enable assert statements.');
```
Comment (chalin) [inline lib/src/command/global_run.dart]: The following seems clearer / more accurate as help text: **Enable runtime checks of assert statements**. WDYT? /cc @kwalrath @kevmoo

File: lib/src/command/run.dart [hunk 2]
```diff
@@ -14,20 +14,20 @@ import '../utils.dart';
...
+  String get name => 'run';
+  String get description => 'Run an executable from a package.';
+  String get invocation => 'pub run <executable> [args...]';
   bool get allowTrailingOptions => false;
 
   RunCommand() {
-    argParser.addFlag("checked",
-        abbr: "c", help: "Enable runtime type checks and assertions.");
-    argParser.addOption("mode", help: "Deprecated option", hide: true);
+    argParser.addFlag('enable-asserts', help: 'Enable assert statements.');
```
Comment (chalin) [inline lib/src/command/run.dart]: Ditto

File: lib/src/executable.dart [hunk 3]
```diff
@@ -22,7 +22,7 @@ import 'utils.dart';
 ///
 /// Arguments from [args] will be passed to the spawned Dart application.
 ///
-/// If [checked] is true, the program is run in checked mode.
+/// If [checked] is true, the program is run with assertions enabled.
```
Comment (chalin) [inline lib/src/executable.dart]: Here and elsewhere: how about "with assertion checking enabled"?

File: lib/src/command/global_run.dart (hunk 1 above)
Comment (natebosch) [reply lib/src/command/global_run.dart]: I took this directly from `dart --help`. If we want to change this I think we should do it in both places.

**Pull request merged**